src/app/workflow/workflow.component.ts

```diff
@@ -65,25 +66,40 @@ export class WorkflowComponent implements OnInit {
     private ngZone: NgZone,
 
   ) {
-    this.route.paramMap.subscribe(paramMap => {
+
+    combineLatest([this.route.paramMap, this.route.queryParamMap])
+      .pipe(map(results => ({paramMap: results[0], queryParamMap: results[1]}))).subscribe(results => {
+
+      console.log("I AM RUNNING THE PARAM MAP SUBSCRIPTION!!!!!")
+
+      let paramMap = results.paramMap
+      let queryParamMap = results.queryParamMap
+
       this.workflowId = parseInt(paramMap.get('workflow_id'), 10);
       let urlTaskId = paramMap.get('task_id');
-      this.api.getWorkflow(this.workflowId).subscribe(
+      let restart = queryParamMap.get('restart')
+
+      // Make a different api call, depending on the information provided.
+      let apiRequest = this.api.getWorkflow(this.workflowId)
+      if(restart) {
+        let clearData = queryParamMap.get('clear_data').toLocaleLowerCase() == 'true'
+        let deleteFiles = queryParamMap.get('delete_files').toLocaleLowerCase() == 'true'
```

Comment:
I am pretty sure we took out the ability to delete files. I think you can only clear data, which also now deletes files. FWIW